utils/visualizer/qviz/content_loader.py

```diff
@@ -2,250 +2,261 @@
-                add_file['tags'] = tags
-                add_files_dict[path] = add_file
-    for entry in log_entry_list:
-        # RemoveFile entry
-        if entry['remove']:
-            path = entry['remove']['path']
-            add_files_dict.pop(path, None)
-
-    return list(add_files_dict.values())
-
-
-def addFiles_from_json_log_files(json_files: list[str], revision_id: str) -> (list[dict], set[str]):
+    try:
+        # Try to find the table in the provided path
+        delta_table = DeltaTable(table_path)
+        return delta_table
+
+    except Exception as e:
+        # If the table doesn't exist, throw an exception
+        print(f"Failed when creating the delta table: {e}\n")
+
+
+# 2. extract metadata from delta table
+# Returns metadata (dict), dimension_count (int) and symbol_count (int)
+def extract_metadata_from_delta_table(
+    delta_table: DeltaTable, revision_id: str
+) -> (dict, int):
     """
-    Extract AddFiles and RemoveFiles from all json log files
-    :param json_files: list, a list of json log file paths
-    :param revision_id: target revisionID
-    :return: a list of AddFile and a set of RemoveFile paths
+    Extract metadata from a Delta table by checking if the revision ID is in that table.
+    If that ID exists in the table, a revision key (e.g. qbeast.revision.1) is created and the metadata is returned.
+    It also calculates the number of symbols used to representate the visualization.
+    :param delta_table: a created delta table.
+    :param revision_id: Configuration entry for the target RevisionID. e.g. 1
+    :return: A dictionary with the metadata related to the revisionId, an integer
+    representing the number of simbols used for the visualization representation.
     """
-    add_files = list()
-    remove_paths = set()
-    for file in json_files:
-        (adds, removes) = load_single_log_file(file, revision_id)
-        add_files.extend(adds)
-        remove_paths = remove_paths.union(removes)
-
-    return add_files, remove_paths
 
+    # List with all the revision ids
+    metadata = delta_table.metadata().configuration #This returns a dictionary with the metadata
+    revision_key = "qbeast.revision." + revision_id
+
+    if revision_key in metadata.keys():
+            metadata_key_str = metadata[revision_key]
+            metadata_key = json.loads(metadata_key_str)
+            dimension_count = len(metadata_key["columnTransformers"])
+            symbol_count = (dimension_count + 5) // 6
+            print(symbol_count)
+            return metadata_key, symbol_count
+    else:
+        print(f"No metadata found for the given RevisionID. There is no Qbeast table.")
+        return None
 
-def load_single_log_file(file_path: str, revision_id: str) -> (list[dict], set[str]):
+# 3. Returns a dictionary, the keys are the cubeids and the values are the actual cubes
+def extract_cubes_from_blocks(delta_table: DeltaTable, symbol_count: int) -> dict:
```

Comment:
Try to find a better name for this method. You are extracting cubes from the table, and you are not providing blocks as inputs.